### PR TITLE
Taking coloring seriously

### PR DESCRIPTION
**Pull request by matbesancon**
Fixing minor style issues and doc typos, documenting coloring and adding in docs

### CODE REVIEW

**Comment by matbesancon** [comment]
The happy thing is that the `coloring` struct was not exported so this is not breaking

File: src/traversals/greedy_color.jl [hunk 1]
```diff
@@ -39,28 +39,27 @@ function perm_greedy_color(g::AbstractGraph, seq::Vector{T}) where T <: Integer
...
 end
 
 
 """
-    random_greedy_color(g, reps=1)
+    random_greedy_color(g, reps)
 
-Color graph `g` iteratively in a random order using a greedy heruistic
-and choose the best coloring out of `reps` such random coloring.
+Colors the graph `g` iteratively in a random order using a greedy heuristic
```
Comment (sbromberger) [inline src/traversals/greedy_color.jl]: We use action verbs to start docstrings. `Color` is correct here.

File: src/traversals/greedy_color.jl [hunk 2]
```diff
@@ -39,28 +39,27 @@ function perm_greedy_color(g::AbstractGraph, seq::Vector{T}) where T <: Integer
...
+    return Coloring{T}(maximum(cols), cols)
 end
 
 """
     degree_greedy_color(g)
 
 Color graph `g` iteratively in the descending order of the degree of the vertices.
 """
-function degree_greedy_color(g::AbstractGraph{T}) where T <: Integer 
+function degree_greedy_color(g::AbstractGraph{T}) where {T <: Integer}
```
Comment (sbromberger) [inline src/traversals/greedy_color.jl]: Why the braces here? It's not consistent with the rest of the file.

**Comment by simonschoelly** [comment]
There is a Jsoc student who will implement some coloring algorithms this summer, I think for solving some matrix problems, so I think it is good that we take this more seriously.

I wonder, if we should rename `color` to `vertexcolor`, in contrast edge colors. It would probably not be a good idea to use a name tuple here instead.

A while ago, there was some discussion here, about a generalized interface for such combinatorial problems, and coloring might be a good candidate to try such a thing.

**Comment by matbesancon** [comment]
I think coloring is usually meant as vertex coloring and is the most common version

File: src/traversals/greedy_color.jl (hunk 2 above)
Comment (matbesancon) [reply src/traversals/greedy_color.jl]: @sbromberger I added braces everywhere for consistency, I find it hard to parse when there is more than just a `T`

File: src/traversals/greedy_color.jl (hunk 1 above)
Comment (matbesancon) [reply src/traversals/greedy_color.jl]: Fixed!

**Comment by matbesancon** [comment]
bump @jpfairbanks @sbromberger for review and merge
